modules/host.go

```diff
@@ -83,49 +93,113 @@ type (
-		// error if its external IP address is unknown. After announcing, the
-		// host will begin accepting contracts.
-		Announce() error
+		// Announce submits a host announcement to the blockchain.  After
+		// announcing, the host will begin accepting contracts.
+		Announce(NetAddress) error
+
+		// ConfigureSmartPrices configures whether the host will automatically
+		// adjust prices based on the network environment. The algorithms aim
+		// to be profit maximizing.
+		ConfigureSmartPrices(contractPricing bool, downloadPricing bool, storagePricing bool, uploadPricing bool) error
 
-		// AnnounceAddress behaves like Announce, but allows the caller to
-		// specify the address announced. Like Announce, this will cause the
-		// host to start accepting contracts.
-		AnnounceAddress(NetAddress) error
+		// ConsistencyCheckAndRepair runs a consistency check on the host,
+		// looking for places where some combination of disk errors, usage
+		// errors, and development errors have led to inconsistencies in the
+		// host. In cases where these inconsistencies can be repaired, the
+		// repairs are made.
+		// TODO: ConsistencyCheckAndRepair() error
 
-		// Capacity returns the amount of storage still available on the
-		// machine. The amount can be negative if the total capacity was
-		// reduced to below the active capacity.
-		Capacity() int64
+		// DeleteSector deletes a sector, meaning that the host will be unable
+		// to upload that sector and be unable to provide a storage proof if
+		// that sector is chosen by the blockchain.
+		DeleteSector(sectorRoot crypto.Hash) error
 
-		// Contracts returns the number of unresolved file contracts that the
-		// host is responsible for.
-		Contracts() uint64
+		// FileContracts returns a list of file contracts that the host
+		// currently has open, along with the volume of data tracked by each
+		// file contract.
+		FileContracts() ([]types.FileContractID, []uint64)
 
-		// DeleteContract deletes a file contract. The revenue and collateral
-		// on the file contract will be lost, and the data will be removed.
-		DeleteContract(types.FileContractID) error
+		// Financials returns the financial statistics of the host.
+		Financials() HostFinancials
 
 		// NetAddress returns the host's network address
 		NetAddress() NetAddress
 
-		// Revenue returns the amount of revenue that the host has lined up,
-		// the amount of revenue the host has successfully captured, and the
-		// amount of revenue the host has lost.
-		//
-		// TODO: This function will eventually include two more numbers, one
-		// representing current collateral at risk, and one representing total
-		// collateral lost.
-		Revenue() (unresolved, resolved, lost types.Currency)
-
 		// RPCMetrics returns information on the types of RPC calls that have
 		// been made to the host.
 		RPCMetrics() HostRPCMetrics
 
+		// SetBandwidthLimits does exactly that.
+		SetBandwidthLimits(altruisticLimits, pricedLimits BandwidthLimits)
```

Comment:
ah whoops. Altruistic limits are the limits for giving away stuff for free. I'm expecting a lot of hosts early on to be comfortable doing X GB / mo of free bandwidth, just to encourage network adoption. Priced limits are the limits when people are paying for the bandwidth. Presumably the limits are going to be much higher when there is payment.